src/elysium/coinsigner.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2020 The Zcoin Core Developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef ZCOIN_ELYSIUM_COINSIGNER_H
+#define ZCOIN_ELYSIUM_COINSIGNER_H
+
+#include "ecdsa_context.h"
+#include "signature.h"
+#include "key.h"
+#include "pubkey.h"
+
+#include <array>
+
+namespace elysium {
+
+class CoinSigner
```

Comment:
Seems like this class is tried to do signing abstraction but it is still tied with `CKey` and `CPubKey`, which is belong to ECDSA with secp256k1. So we have two choices here:

1. Make it right.
2. Remove this abstraction and use `CKey` directly.

For me the second choice is better for now due to it is less likely to that we will have more signing algorithm in the future.